src/nodes/utils/CubeMapNode.js

```diff
@@ -0,0 +1,157 @@
+import TempNode from '../core/TempNode.js';
+import { addNodeClass } from '../core/Node.js';
+import { NodeUpdateType } from '../core/constants.js';
+import { nodeProxy } from '../shadernode/ShaderNode.js';
+import { CubeTexture } from '../../textures/CubeTexture.js';
+import { cubeTexture } from '../accessors/CubeTextureNode.js';
+import CubeRenderTarget from '../../renderers/common/CubeRenderTarget.js';
+import { CubeReflectionMapping, CubeRefractionMapping, EquirectangularReflectionMapping, EquirectangularRefractionMapping } from '../../constants';
+
+const _cache = new WeakMap();
+
+class CubeMapNode extends TempNode {
```

Comment:
@sunag This module should be for internal use only so it is not exported in `Nodes.js`. Still, I'm a bit unsure about the name `CubeMapNode` since its quite similar to `CubeTextureNode`. 

`CubeMapNode` should be in some sense the non-PBR version of `PMREMNode`. Meaning it encapsulate different inputs to provide a single environment map format for the material.

Reply:
I'm not sure if we need this, wouldn't just manipulating the UV be enough since we don't need to pre-filter for non-PBR?

```js
import { texture, equirectUV, reflectVector, refractVector } from 'three/ts';

const coord = reflectVector; // refractVector 
const env = texture( textureEquirec, equirectUV( coord ) );
```

Reply:
The problem with using equirectangular uvs are the artifacts that result at the poles which do not occur with the cube map format. They were previously reported as bugs which was one reason for not directly supporting equirectangular textures in the shader.

Reply:
Interesting, if so let's merge the PR, it seems good to me.